src/TreeMenu/index.tsx

```diff
@@ -63,11 +63,11 @@ class TreeMenu extends React.Component<TreeMenuProps, TreeMenuState> {
     }
   }
 
-  resetOpenNodes = (newOpenNodes?: string[]) => {
+  resetOpenNodes = (newOpenNodes?: string[], key?: string) => {
```

Comment:
Would it be slightly better also have `activeKey` and `focusKey` instead of `key`?